cartographer/mapping/internal/3d/pose_graph/optimization_problem_3d.cc

```diff
@@ -280,7 +280,7 @@ void OptimizationProblem3D::Solve(
 
   // Set the starting point.
   CHECK(!submap_data_.empty());
-  CHECK(submap_data_.Contains(SubmapId{0, 0}));
+  CHECK(submap_data_.SizeOfTrajectoryOrZero(0));
```

Comment:
@ojura, correct that the first submap is fixed, but OptimizationProblem actually finds it properly, so no need to assume it's {0,0} (which is wrong).

Reply:
I have the same understanding. We correctly find the first submap.

Any check that assume trajectory 0 exists is definitely not okay since with life-long mapping every trajectory can possibly be removed.

Reply:
Based on @wohe's suggestion I totally removed the check now.